README.md

```diff
@@ -3,104 +3,83 @@
 - [Overview](#overview)
 - [Requirements](#requirements)
 - [Setup Instructions](#setup-instructions)
+  - [Configure Settings](#configure-settings)
   - [Docker Setup](#steps-using-docker)
   - [Run locally](#steps-to-run-directly)
-  - [Setup Virtual Env](#setup-virtual-env)
-  - [Configure Settings](#configure-settings)
-- [Starting all Processes](#starting-all-processes)
 - [Monitoring and Debugging](#monitoring-and-debugging)
+- [Setting up Slack workflow](#setting-up-slack-workflow)
 - [Architecture Details](#architecture-details)
 
 ## Overview
 
-![Audit-protocol workflow](docs/images/AuditProtocolOverview.png)
-Audit-Protocol is the component of a fully functional, distributed system that works alongside Pooler, off-chain consensus and together they are responsible for
+Architecture overview
+![Architecture](./docs/images/onchain_architecture.png)
+
+Audit-Protocol is the component of a fully functional, distributed system that works alongside Pooler and Epoch Generator together they are all responsible for
 * generating a time series database of changes occurring over smart contract state data and event logs, that live on decentralized storage protocols
 * higher order aggregated information calculated over decentralized indexes maintained atop the database mentioned above
 
-Audit-protocol can also be used for storing stream of json encoded data from any on-chain source(like a smart contract on a blockchain).
+Audit-protocol is used for committing snapshots to on-chain source (like a smart contract on a blockchain) via transaction relayer or directly to it.
```

Comment:
This line is not very clear, maybe we should make it more specific and elaborate a bit more?

Reply:
Improved by adding detailed behaviour